File: src/gpt_generator/gpt_main.py
from gpt_generator.helpers.file_collection import a_create_file_collection

from gpt_generator.modules.file_description_architector import gpt_file_description_architector
from gpt_generator.modules.structure_architector import gpt_project_structure_architector

from models.file_interface import FileCollection, File
from models.gpt_responses_interface import ArchitectorResponse

import logging
import asyncio


logging.basicConfig(level=logging.ERROR)


async def a_gpt_main(prompt: str) -> FileCollection:
  try:
    # Generate initial project and response
    project, response = await gpt_project_structure_architector(prompt)

    # Assert project is of the correct type
    assert isinstance(project, ArchitectorResponse)

    project, response = await gpt_file_description_architector(project, prompt)

    # Assert project is of the correct type
    assert isinstance(project, ArchitectorResponse)

    # Assert response is of the correct type
    assert isinstance(response, str)

    return await a_create_file_collection(
        project,
        response
    )

  except Exception as e:
    # Log any other exceptions
    logging.error(f"An error occurred: {e}")
    return FileCollection("")


def gpt_main(prompt: str) -> FileCollection:
  return asyncio.run(a_gpt_main(prompt))

File: src/gpt_generator/gpt_main.ts
import { aCreateFileCollection } from './helpers/file_collection';
import gpt_file_description_architector from './modules/file_description_architector';
import gpt_project_structure_architector from './modules/structure_architector';
import { createFileCollection, FileCollection } from '../models/file_interface';

const a_gpt_main = async (prompt: string): Promise<FileCollection> => {
  try {
    const [project, _response] = await gpt_project_structure_architector(prompt);

    if (!project || typeof project !== 'object') {
      throw new Error('Project is not of type ArchitectorResponse');
    }

    const [updatedProject, updatedResponse] = await gpt_file_description_architector(project, prompt);

    if (!updatedProject || typeof updatedProject !== 'object' || typeof updatedResponse !== 'string') {
      throw new Error('Invalid project or response type');
    }

    return await aCreateFileCollection(updatedProject, updatedResponse);
  } catch (e) {
    console.error(`An error occurred: ${e}`);
    return createFileCollection('');
  }
};

export const gpt_main = (prompt: string): Promise<FileCollection> => a_gpt_main(prompt);
